### DISCUSSION TITLE
Deleting old messages from queue

### DISCUSSION THREAD

**Discussion by sGoico** [postalserver/postal#2501] [Help with using Postal]
Hello, I found old questions asking this, but I could not find a way to do it yet.

I have old and most likely already delivered messages. I think they were sent but the DB was not working correctly at the time and they are stuck in "Pending" state.

As those emails are old, I'd like to start fresh and delete them. I have tried deleting all pending emails from the messages table inside the "postal-server-1" DB, but that makes postal fail. I have also tried moving their status from "Pending" to "Sent" but that is not enough for them to be removed from the queue.

Is there a way to do this in the DB or GUI? I just want to delete them, I don't have any interest in keeping them even for historical reasons.

BR,
Sebastian

**Comment by willpower232** [comment]
If they're pending I don't think they would have been delivered. You should be good to delete the rows from all tables, not just the queued one. When you say it made Postal fail, what happened exactly?

**Comment by sGoico** [comment]
Hello Will, thanks for answering. To give you more context: I was on the web GUI checking the queue and I deleted all the messages from the DB matching "Pending" state, then I just hit F5, that threw an error on screen and was not able to render it any more (I didn't take a screenshot). As it was an LXC container and I had made a backup minutes before, I just rolled it back and it was up again in 3 minutes.

Then I tested marking them as sent, F5 again ... they were all marked as sent, but still in the queue.

Could you please tell me which other tables I need to delete messages from? Is it the raw tables? I am new to the project and this is my first time playing with the DB :)

Deleting or marking them as sent is good enough for me, I just want to start fresh with an empty queue.

**Reply by willpower232** [reply]
you need to empty the postal.queued-messages as well as removing the relevant entries from postal-server-X.messages
